lib/auth/auth_with_roles.go

```diff
@@ -5828,6 +5828,38 @@ func (a *ServerWithRoles) CloneHTTPClient(params ...roundtrip.ClientParam) (*HTT
 	return nil, trace.NotImplemented("not implemented")
 }
 
+func (a *ServerWithRoles) ExportMaintenanceWindows(ctx context.Context, req proto.ExportMaintenanceWindowsRequest) (proto.ExportMaintenanceWindowsResponse, error) {
+	// Ensure that caller is a teleport server
+	role, ok := a.context.Identity.(authz.BuiltinRole)
+	if !ok || !role.IsServer() {
+		return proto.ExportMaintenanceWindowsResponse{}, trace.AccessDenied("agent maintenance schedule is only accessible to teleport built-in servers")
+	}
+
+	return a.authServer.ExportMaintenanceWindows(ctx, req)
+}
+
+func (a *ServerWithRoles) GetMaintenanceWindow(ctx context.Context) (types.MaintenanceWindow, error) {
+	if err := a.action(apidefaults.Namespace, types.KindMaintenanceWindow, types.VerbRead); err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return a.authServer.GetMaintenanceWindow(ctx)
+}
+
+func (a *ServerWithRoles) UpdateMaintenanceWindow(ctx context.Context, mw types.MaintenanceWindow) error {
```

Comment:
Will this be surfaced in the UI?

Reply:
Cloud users already have a UI for editing the "cloud maintenance start hour", which this resource is automatically derived from.

For on-prem users this will be a cli-only resource for now.